### ISSUE TITLE
Various AWS SSM Providers parameters are missing arguments

### ISSUE THREAD

**Issue by jimsmith** [terraform-providers/terraform-provider-aws#2876]
Hi,

So using latest terraform 0.11.1 following the documentation to create a SSM AWS maintenance window task:
https://www.terraform.io/docs/providers/aws/r/ssm_maintenance_window_task.html

The following is not supported by Terraform, taken from AWS Documentation CLI:
https://docs.aws.amazon.com/cli/latest/reference/ssm/register-task-with-maintenance-window.html
```
[--name <value>]
[--description <value>]
```

These are required so the tasks have meaningful names and descriptions.

Additionally I have noticed that **Comment** & **Timedout (seconds)** there doesn't appear to be any supported arguments to get these populated into AWS

```
TimeoutSeconds
Comment
```
https://docs.aws.amazon.com/systems-manager/latest/userguide/sysman-maintenance-walk.html
https://docs.aws.amazon.com/cli/latest/reference/ssm/update-maintenance-window-task.html 

 
Also I note that when running terraform apply (with no changes) resources get `(new resource required)` which gives the user the 'impression' the resources are being destroyed and then re-created..... however infact the same AWS resources exist and remain..... this is a bug as this shouldn't happen equally when updating parameters shouldn't re-create as AWS has the update maintenance window task.

The below AWS CLI allows me to add in Comment and TimeoutSeconds:
```
aws --profile jim --region eu-west-2 ssm update-maintenance-window-task \
--window-id mw-0a1faf6cb61b24a68 \
--window-task-id 40e3ae25-daae-4b63-9539-2129fde69618 \
--task-invocation-parameters "RunCommand={Comment=A_Comment,TimeoutSeconds=900}"
```

**Comment by jimsmith** [comment]
Adding that description is not supported........ this should be supported.

**Comment by jimsmith** [comment]
Adding that: https://www.terraform.io/docs/providers/aws/r/ssm_maintenance_window_target.html
doesn't have support for **Target Name** nor **description**

**Comment by jimsmith** [comment]
Bump

**Comment by nbrys** [comment]
+1 for this issue

**Comment by egirard78** [comment]
+1 to include those parameters.

**Comment by bryanhodge** [comment]
+1 for this issue

**Comment by bflad** [comment]
Hi @jimsmith (and folks watching this issue) 👋 Thank you for submitting this feature request and sorry about the delays in getting everything sorted here. In the future, its preferable for contributors and maintainers to have bug reports and features requests broken out so the updates can be individually tracked. Many of the items noted in this issue are now resolved and I am going to close this in preference of more targeted GitHub issues.

Releasing in version 2.20.0 of the Terraform AWS Provider later this week:

* resource/aws_ssm_maintenance_window_task: Support resource import and in-place updates
* resource/aws_ssm_maintenance_window_task: Add `task_invocation_parameters` configuration block

Previously released in [version 2.12.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2120-may-24-2019):

* resource/aws_ssm_maintenance_window_target: Add support for `name` and `description` for maintenance window targets

Previously released in [version 1.44.0 of the Terraform AWS Provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1440-november-14-2018):

* resource/aws_ssm_maintenance_window_task: Add `description` and `name` arguments

Tracking for the `aws_ssm_maintenance_window` resource support of `description`: https://github.com/terraform-providers/terraform-provider-aws/issues/9035

For further feature requests or bug reports with functionality within the Terraform AWS Provider, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!
